src/my-element.ts

```diff
@@ -115,6 +115,8 @@ export class MyElement extends LitElement {
   @state()
   save = false
 
+  
+  private cachedImages: { [key: number]: HTMLImageElement } = {}; //Here we define the structure for the cachedImages object: The key will always be a number, the value will always be an HTMLImageElement
```

Comment:
OK so there was an issue when loading the images directly from the url: Every draw would make a new request to load the image. Obviously not good and was actually messing with the positioning of the elements.

We can use this cachedImages object to store all the images that will need to be rendered for this graph. Obviously should not be used for huge numbers of images, but for 2 or 10 or 50 should still be OK.

There are loadImage and getImage methods for storing and accessing images respectively.